audmath/core/api.py

```diff
@@ -193,6 +195,104 @@ def inverse_normal_distribution(
     return _force_float(x)
 
 
+def rms(
+        x: typing.Union[int, float, typing.Sequence, np.ndarray],
+        *,
+        axis: typing.Union[int, typing.Tuple[int]] = None,
+        keepdims: bool = False,
+) -> typing.Union[float, np.ndarray]:
+    r"""Root mean square.
+
+    The root mean square
+    for a signal of length :math:`N`
+    is given by
+
+    .. math::
+
+        \sqrt{\frac{1}{N} \sum_{n=1}^N x_n^2}
+
+    where :math:`x_n` is the value
+    of a single sample
+    of the signal.
+
+    For an empty signal
+    0 is returned.
+
+    Args:
+        x: input signal
+        axis: axis or axes
+            along which the root mean squares are computed.
+            The default is to compute the root mean square
+            of the flattened signal
+        keepdims: if this is set to True,
+            the axes which are reduced
+            are left in the result
+            as dimensions with size one
+
+    Returns:
+        root mean square of input signal
+
+    Example:
+        >>> rms([])
+        0.0
+        >>> rms([0, 1])
+        0.7071067811865476
+        >>> rms([[0, 1], [0, 1]])
+        0.7071067811865476
+        >>> rms([[0, 1], [0, 1]], axis=1)
+        array([0.70710678, 0.70710678])
+
+    """
+    # Don't raise warning for empty signal
+    # when returning NaN
+    x = np.array(x)
+    if x.size == 0:
+        return 0.0
+    return np.sqrt(np.mean(np.square(x), axis=axis, keepdims=keepdims))
+
+
+def rms_db(
```

Comment:
Instead of implementing it as an own function we could also add a `as_db` argument to `rms()`. But I'm also fine with having two functions, but might be easier to maintain if we had only one. On the other hand, I don't think we have to touch those functions ever again once they are implemented.

Reply:
I think it is slightly more convenient to have a separate function as a user, and the other reason is that we can get rid of the `np.sqrt` calculation, which would still be present if you would do `20 * np.log10(rms(signal))`